bolt-sidecar/src/state/execution.rs

```diff
@@ -136,6 +148,11 @@ impl<C: StateFetcher> ExecutionState<C> {
             return Err(ValidationError::BaseFeeTooLow(max_basefee as u128));
         }
 
+        // Validate the chain ID
+        if !req.validate_chain_id(self.chain_id()) {
+            return Err(ValidationError::ChainIdMismatch);
+        }
```

Comment:
can we move this to the top, before `recover_signer()`?
it's a good idea to perform the less expensive checks before the more expensive ones

Reply:
ah yes good point!